### PR TITLE
feat: Migrate ds-controller microservice to EdgeX 3.0

### PR DESCRIPTION
**Pull request by NeethuES-intel**
## PR Checklist

- [X] Added label to the Pull Request for easier discoverability and search
- [X] Commit Message meets guidelines as indicated in the URL https://github.com/intel-retail/automated-vending/blob/main/.github/CONTRIBUTING.md
- [X] Every commit is a single defect fix and does not mix feature addition or changes
- [ ] Unit Tests have been added for new changes
- [ ] Updated Documentation as relevant to the changes
- [X] All commented out code has been removed
- [ ] If you've added a dependency, you've ensured license is compatible with repository license and clearly outlined the added dependency.
- [ ] PR change contains code related to security
- [ ] PR introduces changes that breaks compatibility with other modules (If YES, please provide details below)

## Issue this PR will close
close: #134 

## Test Instructions if applicable
Run the following make commands from the ds-controller directory, all should run without erros

- make docker
- make test

### CODE REVIEW

**Review by jim-wang-intel — APPROVED** [review]
LGTM

**Review by lenny-intel — APPROVED** [review]
LGTM

**Review by lenny-intel — CHANGES_REQUESTED** [review]
Still need to convert the res/devices/device-list.toml to yaml

File: ds-controller-board/res/devices/device-list.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+# SPDX-License-Identifier: BSD-3-Clause
+
+# Pre-define Devices
+deviceList:
+  dame: controller-board
+  profileName: ds-controller-board
+  protocols: 
+    autoEvents:
+      interval: 3s
+      onChange: true
```
Comment (lenny-intel) [inline ds-controller-board/res/devices/device-list.yaml:11]: Indentation in TOML we off . I matter in YAML, but not in TOML.
```suggestion
  protocols: 
  autoEvents:
    interval: 3s
    onChange: true
```

**Review by lenny-intel — CHANGES_REQUESTED** [review]

**Pull request merged**